### PR TITLE
feat(ecocredit/marketplace): AllowAskDenom parameter

### PR DESCRIPTION
**Pull request by technicallyty**
## Description

Closes: #624
Closes: #856

Adds AllowAskDenom parameter and verifies AskDenom is allowed when creating a sell order.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/ecocredit/server/marketplace/utils.go [hunk 1]
```diff
@@ -3,31 +3,21 @@ package marketplace
...
+	"github.com/cosmos/cosmos-sdk/types"
+)
 
 // isDenomAllowed checks if the denom is allowed to be used in orders.
-func isDenomAllowed(ctx context.Context, store api.StateStore, denom string) (bool, error) {
-	return store.AllowedDenomTable().Has(ctx, denom)
+func isDenomAllowed(ctx context.Context, denom string, pk ecocredit.ParamKeeper) bool {
+	sdkCtx := types.UnwrapSDKContext(ctx)
+	var params core.Params
+	pk.GetParamSet(sdkCtx, &params)
```
Comment (ryanchristo) [inline x/ecocredit/server/marketplace/utils.go:16]: I think this is fine but we unwrap sdk context twice where this is used. We could pass sdk context as a param, move these lines outside the helper function, or leave it as is. No strong preference.

**Review by ryanchristo — APPROVED** [review]
Looks good to me.

File: x/ecocredit/server/marketplace/utils.go (hunk 1 above)
Comment (technicallyty) [reply x/ecocredit/server/marketplace/utils.go]: changed the func parameter to sdk.Context https://github.com/regen-network/regen-ledger/pull/969/commits/8ecf489e45bf6907ca446a256bbc996a0b6fcfb7

**Review by aleem1314 — APPROVED** [review]
lgtm

**Pull request merged**